meta-client/src/main/java/org/apache/helix/metaclient/recipes/lock/LockClientInterface.java

```diff
@@ -0,0 +1,62 @@
+package org.apache.helix.metaclient.recipes.lock;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.metaclient.api.MetaClientInterface;
+
+public interface LockClientInterface {
+  /**
+   * Acquires a lock by creating a node at entry key and lockinfo info.
+   * Will fail and return False if path and lockinfo is invalid.
+   * @param key key to identify the entry
+   * @param info Metadata of the lock
+   * @param mode EntryMode identifying if the entry will be deleted upon client disconnect
+   * @return True if the lock is acquired. False if failed to acquire (catches exception).
+   */
+  boolean acquireLock(String key, LockInfo info, MetaClientInterface.EntryMode mode);
+
+  /**
+   * Renews lock for a TTL Node.
+   * Will fail if key is an invalid path or isn't of type TTL.
+   * @param key key to identify the entry
+   * @return True if the lock was renews. False if failed to renew (catches exception).
+   */
+  boolean renewTTLLock(String key);
+
+  /**
+   * Releases the lock by deleted the node at entry key.
+   * Will fail if key is an invalid path.
+   * @param key key to identify the entry
+   * @return True if the lock was released. False if failed to release (catches exception).
+   */
+  boolean releaseLock(String key);
```

Comment:
Is this method idempotent? What will happen if user release a lock multiple times?

Reply:
+1 on qqu's comment.
Probably we want more detail regarding what does "invalid path" mean here. 
When releasing an lock doe second time, do we want to differentiate already released lock or non-existing lock? It is hard to do when underlying datastore does remove lock entry when released.

Reply:
@qqu0127 This I think depends on the implementation of release lock, though the expected behavior of releasing a lock on one that either doesn't exist or already has been released should error. As per the design, releaslock releases the lock using transactional support to avoid race conditions.

@xyuanlu I think this is related to @mgao0 question above on creating a separate class to surface exceptions/errors. Depending on the implementation, that will or will not differentiate between those two errors (though I think it is a good idea to too). With a separate error/result wrapper class we can capture this error and return it, regardless of the underlying datastore.

Reply:
@Marcosrico thanks for the reply.
Yes, there could be different implementations. But the idea is, we are now defining the interface, thus now is the time we define the behavior and semantics.
Back to the `releaseLock` behavior, idempotent API will have some benefits for simplifying logic especially in case of retry. It could be that, as long as the lock has been released, it returns true or success. 
IMO, it's just one way to define the behavior, and there are tradeoff. But the bottomline is we should document the expected behavior clearly (or clearly say it's completely up to the impl). We can discuss more on this 😄